drivers/ruby/lib/net.rb

```diff
@@ -455,6 +457,7 @@ def initialize(opts={})
       @auth_key = opts[:auth_key] || ""
       @timeout = opts[:timeout].to_i
       @timeout = 20 if @timeout <= 0
+      @ssl_opts = opts.reject { |k, v| !k.to_s.start_with?('ssl') } if opts.is_a?(Hash)
```

Comment:
It looks like this will set `@ssl_opts` to `nil` if `!opts.is_a?(Hash)`, but in `init_socket` we call `@ssl_opts.empty?` without checking whether or not it's `nil`.